src/chains/base.ts

```diff
@@ -1,12 +1,45 @@
 import { Chain, ChainContract } from 'viem'
 import { base as viemBase } from 'viem/chains'
-import { opStackL2ContractAddresses } from '../types/opStackContracts'
+
+export type OpStackChain = Chain & {
+  contracts: OpStackL2Contracts
+  optimismConfig: OptimismConfig
+}
+
+export type OptimismConfig = {
```

Comment:
In it's final and complete form an optimism config looks like this (forgive the mix of json and yaml)

```
base_goerli
chain_id: 84531
public_rpc: https://goerli.base.org
sequencer_rpc: https://goerli-sequencer.base.org
explorer: https://goerli.basescan.org

system_config_addr: "0xb15eea247ece011c68a614e4a77ad648ff495bc1"
batch_inbox_addr: "0x8453100000000000000000000000000000000000"

genesis:
  l1:
    hash: "0x73d89754a1e0387b89520d989d3be9c37c1f32495a88faf1ea05c61121ab0d19"
    number: 8410981
  l2:
    hash: "0xa3ab140f15ea7f7443a4702da64c10314eb04d488e72974e02e2d728096b4f76"
    number: 0
  l2_time: 1675193616
contracts:
    "AddressManager": "0x4Cf6b56b14c6CFcB72A75611080514F94624c54e",
  "L1CrossDomainMessengerProxy": "0x8e5693140eA606bcEB98761d9beB1BC87383706D",
  "L1ERC721BridgeProxy": "0x5E0c967457347D5175bF82E8CCCC6480FCD7e568",
  "L1StandardBridgeProxy": "0xfA6D8Ee5BE770F84FC001D098C4bD604Fe01284a",
  "L2OutputOracleProxy": "0x2A35891ff30313CcFa6CE88dcf3858bb075A2298",
  "OptimismMintableERC20FactoryProxy": "0xa88530E2DD811363cA3Ef479dBab3C0BF73d90b1",
  "OptimismPortalProxy": "0xe93c8cD0D409341205A592f8c4Ac1A5fe5585cfA",
  "ProxyAdmin": "0xbc0Fc544736b7d610D9b05F31B182C8154BEf336"
```
I think seperate the contracts into l1 and l2 makes sense though. But I would key contract: l1 instead of l1: contracts so you stay consistent with other OP configs

Reply:
l2 contracts will just go in the normal chain.contracts

Reply:
ah clicked what you are going for. It's an l2Chain object that has an l1 object that itself is a chain object